### PR TITLE
MCXN94X FlexPWM Enablement

### PR DESCRIPTION
**Pull request by EmilioCBen**

### CODE REVIEW

**Review by decsny — DISMISSED** [review]

File: boards/nxp/frdm_mcxn947/frdm_mcxn947_mcxn947_cpu0.dts [hunk 1]
```diff
@@ -116,3 +120,9 @@
 &enet_mdio {
 	status = "okay";
 };
+
+&flexpwm1_pwm0 {
+	status = "okay";
+	pinctrl-0 = <&pinmux_flexpwm1_pwm0>;
+	pinctrl-names = "default";
+};
```
Comment (mmahadevan108) [inline boards/nxp/frdm_mcxn947/frdm_mcxn947_mcxn947_cpu0.dts:128]: Similar to other peripherals, have the pinctrl defines in the common dtsi file and enable the peripheral in the cpu0 file.

File: dts/arm/nxp/nxp_mcxn94x_common.dtsi [hunk 2]
```diff
@@ -603,6 +603,50 @@
...
+		reg = <0xd0000 0x1000>;
+		interrupt-names = "RELOAD-ERROR", "FAULT";
+		interrupts = <118 0>, <119 0>;
+		flexpwm1_pwm0: pwm0 {
+			compatible = "nxp,imx-pwm";
+			index = <0>;
+			interrupt-names = "CAPTURE-COMPARE-RELOAD-SUB0";
+			interrupts = <120 0>;
+			#pwm-cells = <3>;
+			clocks = <&syscon MCUX_BUS_CLK 0 0>;
```
Comment (mmahadevan108) [inline dts/arm/nxp/nxp_mcxn94x_common.dtsi:618]: please delete the last 2 cells. `syscon` does not use it.

File: dts/arm/nxp/nxp_mcxn94x_common.dtsi [hunk 3]
```diff
@@ -603,6 +603,50 @@
...
+			nxp,prescaler = <128>;
+			status = "disabled";
+			run-in-wait;
+		};
+
+		flexpwm1_pwm2: pwm2 {
+			compatible = "nxp,imx-pwm";
+			index = <2>;
+			interrupt-names = "CAPTURE-COMPARE-RELOAD-SUB2";
```
Comment (mmahadevan108) [inline dts/arm/nxp/nxp_mcxn94x_common.dtsi:639]: Is this name used by the driver?

**Review by mmahadevan108 — CHANGES_REQUESTED** [review]
Please look at comments.

File: boards/nxp/frdm_mcxn947/frdm_mcxn947_mcxn947_cpu0.dts (hunk 1 above)
Comment (EmilioCBen) [reply boards/nxp/frdm_mcxn947/frdm_mcxn947_mcxn947_cpu0.dts:126]: I understand now, please check again to see if this is correct.

File: dts/arm/nxp/nxp_mcxn94x_common.dtsi [hunk 4]
```diff
@@ -605,7 +605,8 @@
 		enet_mdio: mdio {
 			#address-cells = <1>;
 			#size-cells = <0>;
-			compatible = "nxp,enet-qos-mdio";
+			c
```
Comment (mmahadevan108) [inline dts/arm/nxp/nxp_mcxn94x_common.dtsi:608]: Some typo here

File: dts/arm/nxp/nxp_mcxn94x_common.dtsi [hunk 5]
```diff
@@ -618,6 +619,107 @@
 		clk-divider = <1>;
 	};
 
+	flexpwm0: flexpwm@ce000 {
+		compatible = "nxp,flexpwm";
+		reg = <0xce000 0x1000>;
+		interrupt-names = "RELOAD-ERROR", "FAULT";
```
Comment (mmahadevan108) [inline dts/arm/nxp/nxp_mcxn94x_common.dtsi:625]: NO need for this property.

File: boards/nxp/frdm_mcxn947/frdm_mcxn947_mcxn947_cpu0.dts (hunk 1 above)
Comment (mmahadevan108) [reply boards/nxp/frdm_mcxn947/frdm_mcxn947_mcxn947_cpu0.dts:139]: Please fix a typo in the commit message. Change `inlcuding ` to `including`

**Review by decsny — DISMISSED** [review]

**Review by mmahadevan108 — CHANGES_REQUESTED** [review]
Please fix a typo in the commit message. Change `inlcuding`  to `including`

**Comment by EmilioCBen** [comment]
> Please fix a typo in the commit message. Change `inlcuding` to `including`

Thank you, typo fixed.

**Review by decsny — APPROVED** [review]

**Review by mmahadevan108 — APPROVED** [review]

**Review by dleach02 — APPROVED** [review]

**Pull request merged**